emcc.py

```diff
@@ -2412,9 +2412,11 @@ def phase_post_link(options, in_wasm, wasm_target, target):
   settings.TARGET_BASENAME = unsuffixed_basename(target)
 
   if options.oformat in (OFormat.JS, OFormat.MJS):
-    settings.TARGET_JS_NAME = target
+    settings.TARGET_JS_PATH = target
   else:
-    settings.TARGET_JS_NAME = get_secondary_target(target, '.js')
+    settings.TARGET_JS_PATH = get_secondary_target(target, '.js')
```

Comment:
In general we use settings for global things that need to be shared between python and JS library code, or at least across multiple part of the compiler.   For things that are local to emcc.py we historically have just use locals, but we these days we have the EmccState object.   Just make this `state.js_target` and just that below rather then `js_target` local.  

Also, the fact that we could have landed this means we are missing test coverage.   Let me take a look if it makes sense to extends and existing test or add a new one.

Reply:
We don't have almost any EXPORT_ES6 coverage, def not execution one, so it wouldn't have caught this. That's a long-standing issue :(

Now that I know about this specific issue, I can add a simple regex regression test, but the real problem is catching them in advance.

Reply:
Added a test too.